### PR TITLE
Fix pop after reset

### PR DESCRIPTION
**Pull request by andreeaflorescu**
### Summary of the PR

This PR adds a (partial) fix for a fuzzing finding. The finding is low (generates a timeout when running with a timeout of 60 seconds), and somewhat similar to https://github.com/rust-vmm/vm-virtio/pull/196.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by rbradford — DISMISSED** [review]

**Comment by andreeaflorescu** [comment]
@rbradford I had to do a push force to fix a test. Can you take another look?

**Review by rbradford — DISMISSED** [review]

File: crates/virtio-queue/src/queue.rs [hunk 1]
```diff
@@ -1500,4 +1506,45 @@ mod tests {
...
+            Descriptor::new(76842670169653248, 1114115, 0, 0),
+            Descriptor::new(16, 983040, 126, 3),
+            Descriptor::new(897648164864, 0, 0, 0),
+            Descriptor::new(111669149722, 0, 0, 0),
+        ];
+        vq.build_multiple_desc_chains(&descriptors).unwrap();
+
+        let mut q: Queue = vq.create_queue().unwrap();
+        q.reset();
+        q.set_ready(true);
```
Comment (lauralt) [inline crates/virtio-queue/src/queue.rs:1535]: Should we add a test for the reverse, i.e. set_avail_addr is setting the available ring address != 0, but the ready flag is false?

Comment (andreeaflorescu) [reply crates/virtio-queue/src/queue.rs:1535]: I can extend this test to account for that.

Comment (andreeaflorescu) [reply crates/virtio-queue/src/queue.rs:1537]: Extended the test! @lauralt @rbradford please re-review.

**Review by lauralt — APPROVED** [review]

**Review by rbradford — APPROVED** [review]

**Pull request merged**